tools/make_metadata.py

```diff
@@ -76,10 +82,13 @@
         if args.sign:
             assert conf.key_file
             assert conf.cert_file
-            eid, xmldoc = sign_entity_descriptor(eid, args.id, secc)
+            sign_alg = args.signalg
+            digest_alg = args.digestalg
+            eid, xmldoc = sign_entity_descriptor(eid, args.id, secc, sign_alg,
+                                                 digest_alg)
         else:
             xmldoc = None
 
         valid_instance(eid)
-        xmldoc = metadata_tostring_fix(eid, nspair, xmldoc)
+        xmldoc = metadata_tostring_fix(eid, nspair, xmldoc.encode())
```

Comment:
hm, this is fishy. I've seen `metadata_tostring_fix` before and I think it should be changed to check the type. That means, you should not need to `.encode` the given doc.

Reply:
Hey @c00kiemon5ter, maybe something like that? https://github.com/psmiraglia/pysaml2/commit/dfe523e586d79b8d888d1604dd70bcc2994b90e5

Reply:
sorry for have made a mess with commits history, my fault!
Anyway, this conversation is solved now